connector.rb

```diff
@@ -1,37 +1,2787 @@
+        {
+          'name' => "root_#{field_name}",
+          'type' => :object,
+          'properties' => eis
+        }
+      ]
+      field_list = call(
+        'build_query_field_list',
+        wrapped_input,
+        wrapped_eis,
+        eos
+      )
+      "#{operation_type} #{field_list}"
+    end,
+
+    action_execute: lambda do |connection, input, eis, eos, operation_type|
+      field_name = input['object']
+      unless field_name.blank?
+        is_object = field_name.ends_with?('{}')
+
+        query = call(
+          'build_full_query',
+          field_name,
+          input,
+          eis,
+          eos,
+          operation_type
+        )
+        result = call(
+          'run_gql',
+          connection,
+          query,
+          nil,
+          nil,
+          input['account']
+        )
+        if is_object
+          field_name = field_name[0..-3]
+          result = result[field_name]
+        end
+        result
+      end
+    end
   },
 
-  object_definitions: {
+  triggers: {
+    new_event: {
+      title: 'Webhook event',
+      subtitle: 'Triggers when a selected 4me object, e.g person, is created/updated, ' \
+                'or on an automation rule notification.',
+      description: lambda do |input, picklist_label|
+        "New <span class='provider'>webhook</span> in <span class='provider'>4me</span>"
+      end,
+
+      help: lambda do |input, picklist_label, connection, webhook_base_url|
+        <<~HTML
+          Creates a job when an event is received from 4me. To set this webhook up,
+          you will need to register the webhook below in 4me under "settings" => "webhooks". <br><br>
+          <b>Webhook endpoint URL</b>
+          <b class="tips__highlight">#{webhook_base_url}</b>
+          More information on how to use 4me automation rules and webhooks can be found on the <a href="https://developer.4me.com//v1/workato_connector/" target="_blank">4me developer pages</a>.
+        HTML
+      end,
+
+      config_fields:
+        [
+          {
+            name: 'event_selection',
+            label: 'Event',
+            control_type: 'select',
+            pick_list: 'webhook_events',
+            optional: false,
+            hint: 'Select a specific event.'
+          },
+          {
+            name: 'webhook_identifier',
+            label: 'Webhook identifier',
+            optional: false,
+            hint: 'The webhook GraphQL identifier. An array of GraphQL identifiers can be set via "Formula" mode.'
+          },
+          {
+            name: 'webhook_policy',
+            label: 'JWT algorithm',
+            control_type: 'select',
+            pick_list: 'webhook_jwt',
+            optional: false,
+            hint: 'The webhook policy JWT algorithm.'
+          },
+          {
+            ngIf: 'input.webhook_policy == "HS256" || input.webhook_policy == "HS384" || ' \
+                  'input.webhook_policy == "HS512"',
+            name: 'webhook_secret_hs',
```

Comment:
It is preferable to have full `hmac` insted of `hs`

Reply:
Updated, will be in the next commit.